### PR TITLE
format_entries sorts directories before files

### PR DESCRIPTION
**Pull request by jrwdunham**

### CODE REVIEW

File: browse-helpers.js [hunk 1]
```diff
@@ -16,7 +16,10 @@ export function decode_browse_response(response) {
 };
 
 export function format_entries(data, parent_path, parent = null) {
-  return data.entries.map(element => {
+  var directories = [];
```
Comment (sevein) [inline browse-helpers.js]: I think you want to use the new `let`. It wouldn't make a difference here but I think it's the new practice?

File: browse-helpers.js [hunk 2]
```diff
@@ -16,7 +16,10 @@ export function decode_browse_response(response) {
...
+  var files = [];
+  for (var index = 0; index < data.entries.length; ++index) {
+    var element = data.entries[index];
```
Comment (sevein) [inline browse-helpers.js]: And here `let` makes more sense to me.

**Comment by sevein** [comment]
Minor observation about `let` vs `var` but 👍

**Comment by jrwdunham** [comment]
@sevein yay for block scope

**Comment by sevein** [comment]
👍

**Pull request merged**